crates/dbsp/src/trace/cursor/mod.rs

```diff
@@ -123,22 +150,29 @@ pub trait Cursor<K, V, T, R> {
     /// Moves the cursor to the previous key.
     fn step_key_reverse(&mut self);
 
-    /// Advances the cursor to the specified key.
+    /// Advances the cursor to the specified key.  If `key` itself is not
+    /// present, advances to the first key greater than `key`; if there is no
+    /// such key, the cursor becomes invalid.
+    ///
+    /// This has no effect if the cursor is already positioned past `key`, so it
+    /// might be desirable to call [`rewind_keys`](Self::rewind_keys) first.
     fn seek_key(&mut self, key: &K);
 
-    /// Move the cursor to the first key that satisfies `predicate`.
+    /// Advances the cursor to the first key that satisfies `predicate`.
     /// Assumes that `predicate` remains true once it turns true.
     fn seek_key_with<P>(&mut self, predicate: P)
     where
         P: Fn(&K) -> bool + Clone;
 
-    /// Move the cursor back to the first key that satisfies `predicate`.
+    /// Move the cursor backward to the first key that satisfies `predicate`.
```

Comment:
We should also document the fact that a cursor can only move in one direction at any given time. A cursor is initialized with the forward direction, i.e., only `step_key` and `seek_key` methods can be invoked, but not `step_key_reverse`, `seek_key_reverse`.  The direction can be reversed using `fast_forward_keys` to re-start iteration in reverse from the last keys, and `rewind_keys` to restart iteration in reverse from the last key and `fast_forward_keys` to restart iteration forward from the first key. 

Likewise, whenever the cursor moves to a new key, it is set to iterate over values for this key in the forward direction and can be reversed using `fast_forward_vals` and `rewind_vals`.

I should have documented all this myself, sorry about the omission.

Reply:
Wow, that's more complicated than I realized.

I pushed a revision that explains it.